tests/runner/src/tests/smoke.rs

```diff
@@ -19,39 +19,15 @@ use std::sync::Arc;
 
 #[fluvio_test]
 pub async fn run(client: Arc<Fluvio>, option: TestCase) {
```

Comment:
Idea will be 
```
async fn run(client: Arc<Fluvio>, option: TestCase<SmokeTestOption>) {
....
}
```
or alternatively:
```
async fn run(client: Arc<Fluvio>, option: TestCase) {
let smoke_option = option.into();
}
```



`SmokeTestOption` can be any struct that complies with `StructOpt`.   With that, can restore previous way to define SmokeTestOption:
```
pub struct SmokeTestOptions {
    /// number of produce iteration for a single producer
    #[structopt(short, long, default_value = "1")]
    pub produce_iteration: u16,

    // record size
    #[structopt(long, default_value = "100")]
    pub record_size: usize,

    // number of parallel producer
    #[structopt(long, default_value = "1")]
    pub producer_count: u16,
}
```

Can can set option similar like `cargo test`, where you can specify any extra arguments
```
$ flv-test smoke --local  --  --producer.iteration=10
```
and can even show help:
```
```
$ flv-test smoke --local  --help

```
```

Reply:
I have the latter partially implemented in tests/runner/src/utils/smoke/mod.rs.

The only part that is missing is the `StructOpt` piece, which will be a little more complicated than we initially planned. I'll try to give it a go to get rid of the `HashMap<String, String>`, but I believe this is a good and usable improvement for custom vars.

Reply:
Doesn't seems to be complicated:  https://docs.rs/structopt/0.3.21/structopt/trait.StructOpt.html#method.from_iter.

Nice thing about keeping `StructOpt` is that we can provide defaults, env var, etc.